### PR TITLE
Shahryart/additional pagination key error check

### PR DESCRIPTION
**Pull request by shahryart**
add error checks for key field

### CODE REVIEW

File: graphql/schemabuilder/pagination.go [hunk 1]
```diff
@@ -303,6 +303,27 @@ func (funcCtx *funcContext) consumePaginatedArgs(sb *schemaBuilder, in []reflect
...
+	}
+	nodeKey := reverseGraphql(nodeObj.key)
+	if nodeKey == "" {
+		return nodeKey, fmt.Errorf("a key field must be registered for paginated objects")
+	}
+	if nodeType.Kind() == reflect.Ptr {
+		nodeType = nodeType.Elem()
+	}
+	if _, ok := nodeType.FieldByName(nodeKey); !ok {
+		return nodeKey, fmt.Errorf("field doesn't exist on struct")
```
Comment (shahryart) [inline graphql/schemabuilder/pagination.go]: this check seems redundant? reflect will fail before this

Comment (stephen) [reply graphql/schemabuilder/pagination.go]: is there a test verifying that behavior?

File: graphql/schemabuilder/reflect.go [hunk 2]
```diff
@@ -56,6 +56,18 @@ func makeGraphql(s string) string {
 	return b.String()
 }
 
+func reverseGraphql(s string) string {
```
Comment (stephen) [inline graphql/schemabuilder/reflect.go]: this feels too vague as-is (as is `makeGraphql`..)

`reverseGraphqlFieldName`? `graphqlToGoName`?

File: graphql/schemabuilder/types.go [hunk 3]
```diff
@@ -61,6 +61,16 @@ func (s *Object) FieldFunc(name string, f interface{}, options ...FieldFuncOptio
 	s.Methods[name] = m
 }
 
+// Key registers the key field on an object. The field should be specified with the initial
```
Comment (stephen) [inline graphql/schemabuilder/types.go]: how about "should be specified by the name of the graphql field, i.e. camelCased for struct fields"

File: graphql/schemabuilder/types.go [hunk 4]
```diff
@@ -61,6 +61,16 @@ func (s *Object) FieldFunc(name string, f interface{}, options ...FieldFuncOptio
...
+// lowercased.
+// For example, for an object User:
+// type struct User {
+//	 Id int64
+// }
+// The object is registered as:
+// object := schema.Object("user", User{})
```
Comment (stephen) [inline graphql/schemabuilder/types.go]: maybe remove lines 70/71

File: graphql/schemabuilder/pagination.go (hunk 1 above)
Comment (shahryart) [reply graphql/schemabuilder/pagination.go]: can't be tested with the current reflect check but nice to leave it in

**Pull request merged**